comp110-worksheet-6/OxoBoard.cs

```diff
@@ -6,53 +6,272 @@
 
 namespace comp110_worksheet_6
 {
-	public enum Mark { None, O, X };
+	public enum Mark 
+	{ 
+		None, 
+		O, 
+		X 
+	};
 
 	public class OxoBoard
 	{
 		// Constructor. Perform any necessary data initialisation here.
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
-		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+
+		private Mark[,] board;
+        private int itemsInARow;
+
+		public OxoBoard(int width = 3, int height = 3, int inARow = 3)
+        {
+			// Initialises game board to a 3 * 3 square
+			board = new Mark[width, height];
+            itemsInARow = inARow;
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			// Returns the boards value, if the value is null then it returns Mark.None
+			return board[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if (board[x, y] == Mark.None)
+			{
+				board[x, y] = mark;
+				return true;
+
+			} 
+			else 
+			{
+				return false;
+			}
+
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			for (int x = 0; x < board.GetLength(0); x++)
+			{
+				for (int y = 0; y < board.GetLength(1); y++)
+				{
+					if (board[x, y] == Mark.None) 
+					{
+						return false;
+					}
+				}
+			}
+
+			return true;
 		}
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        private bool CheckDiagonals(int x, int y, Mark targetMark)
+        {
+            bool checkLeft = true;
+            bool checkRight = true;
+            bool winCondition = false;
+            // Check down and right diagonals.
+            // Only need to check downwards diagonals to still check every possibility.
+            int testX = x;
+            int testY = y;
+            testX++;
+            testY++;
+            if (testX < board.GetLength(0) && testY < board.GetLength(1))
+            {
+                if (board[testX, testY] == targetMark)
+                {
+                    int symbolCount = 2;
```

Comment:
This function is in need of some more comments -- why `symbolCount = 2` here for example?